src/frontend/js/widgets/SyllabusCourseRunsList/index.spec.tsx

```diff
@@ -614,4 +633,75 @@ describe('<SyllabusCourseRunsList/>', () => {
     // Assert the second course run to not be wrapped in a link.
     expect(within(listElements[1]).queryByRole('link')).not.toBeInTheDocument();
   });
+
+  it('renders ongoing course runs with enrollment information', async () => {
+    const user = UserFactory().one();
+    const course = CourseLightFactory().one();
+
+    const onGoingCourseRun = CourseRunFactoryFromPriority(Priority.ONGOING_CLOSED)({
+      title: 'Ongoing course run',
+      resource_link: 'https://openedx.endpoint/course-v1:edX+DemoX+Session2/info/',
+    }).one();
+
+    const courseRuns: CourseRun[] = [
+      onGoingCourseRun,
+      CourseRunFactoryFromPriority(Priority.ARCHIVED_CLOSED)({
+        title: 'Closed course run',
+        resource_link: 'https://openedx.endpoint/course-v1:edX+DemoX+Session1/info/',
+      }).one(),
+      CourseRunFactoryFromPriority(Priority.TO_BE_SCHEDULED)({
+        title: 'To be scheduled course run',
+        resource_link: 'https://openedx.endpoint/course-v1:edX+DemoX+Session4/info/',
+      }).one(),
+      CourseRunFactoryFromPriority(Priority.FUTURE_NOT_YET_OPEN)({
+        title: 'Future course run',
+        resource_link: 'https://openedx.endpoint/course-v1:edX+DemoX+Session3/info/',
+      }).one(),
+    ];
+
+    courseRuns.forEach((courseRun) => {
+      fetchMock.get(
+        `https://demo.endpoint/api/enrollment/v1/enrollment/${user.username},${courseRun.resource_link}`,
+        {
+          is_active: true,
+        },
+      );
+    });
+
+    render(
+      <SyllabusCourseRunsList
+        courseRuns={courseRuns}
+        course={course}
+        maxArchivedCourseRuns={MAX_ARCHIVED_COURSE_RUNS}
+      />,
+      {
+        wrapper: ({ children }) => <Wrapper user={user}>{children}</Wrapper>,
+      },
+    );
+
+    const portalContainer = getPortalContainer();
+    expectCourseRunInList(portalContainer, courseRuns[0]);
+    expectCourseRunInList(portalContainer, courseRuns[1]);
+    expectCourseRunInList(portalContainer, courseRuns[2]);
+    expectCourseRunInList(portalContainer, courseRuns[3]);
+
+    const listElements = screen.getAllByRole('listitem');
+    expect(listElements.length).toBe(4);
+
+    // Assert there is only one link, one label "Enrolled" and one request to retrieve enrollment.
+    await waitFor(() => {
+      expect(screen.getAllByText('Enrolled').length).toBe(1);
+    });
+
+    const links = screen.getAllByRole('link');
+    expect(links.length).toBe(1);
+    expect(fetchMock.calls().length).toBe(1);
+
+    // Assert user's enrollment state has been checked for the ongoing course run.
+    const link = within(portalContainer).getByRole('link');
+    expect(link.getAttribute('href')).toBe(courseRuns[0].resource_link);
+    expect(fetchMock.calls()[0][0]).toBe(
+      `https://demo.endpoint/api/enrollment/v1/enrollment/${user.username},${onGoingCourseRun.resource_link}`,
+    );
```

Comment:
in other places i did stuff like : 
```js
const calledUrls = fetchMock.calls().map((call) => call[0]);
expect(calledUrls).toHaveLength(nbApiCalls);
expect(calledUrls).toContain(
  `https://demo.endpoint/api/enrollment/v1/enrollment/${user.username},${onGoingCourseRun.resource_link}`,
);
```
Because the calls order doesn't matter, i prefer this way, it avoid refactor in case of new api calls in the future.

Reply:
Much better indeed !